spek-runtime/src/commonMain/kotlin/org/spekframework/spek2/runtime/Executor.kt

```diff
@@ -70,26 +68,38 @@ class Executor {
                     }
                     is TestScopeImpl -> {
                         val exception = withContext(scopeCoroutineContext) {
-                            val job = launch {
+                            val job = async {
                                 scope.before()
                                 scope.invokeBeforeTestFixtures()
                                 scope.execute()
                             }
 
                             if (scope.timeout == 0L) {
                                 try {
-                                    job.join()
+                                    job.await()
```

Comment:
`await` allows us to capture the actual exception thrown compared to `launch` which will propagate the exception to the parent context.